Source/AwsCommonRuntimeKit/sdkutils/endpoint/EndpointsResolvedEndpoint.swift

```diff
@@ -0,0 +1,87 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCSdkUtils
+import Foundation
+
+/// Resolved endpoint
+public class CRTAWSEndpointResolvedEndpoint {
+    let rawValue: OpaquePointer
+
+    /// Initialize a new resolved endpoint
+    /// - Parameter rawValue: The raw value of the resolved endpoint
+    init(rawValue: OpaquePointer) {
+        self.rawValue = rawValue
+    }
+
+    /// Get the type of the resolved endpoint
+    /// - Returns: The type of the resolved endpoint
+    public func getType() -> CRTAWSEndpointsResolvedEndpointType {
+        let type = aws_endpoints_resolved_endpoint_get_type(rawValue)
+        return CRTAWSEndpointsResolvedEndpointType(rawValue: type)
+    }
+
+    /// Get the URL of the resolved endpoint
+    /// - Returns: The URL of the resolved endpoint
+    public func getURL() throws -> String? {
```

Comment:
should this be `throws -> String` instead of `throws -> String?`

in what circumstance do we not-error but also not-return-a-string?

Reply:
also, we should probably document the circumstances under which errors might be thrown. The header files say:
> Valid only if resolved endpoint has endpoint type and will error otherwise.

but I'm not 100% sure what that means, not being super familiar with all this...

is it possible for an endpoint to be just totally invalid and every function on it fails?
why bother creating an object that's invalid? why not just have its creation function report failure?

Reply:
it might be good to explain here that result of resolution is either endpoint or error. errors only define message, so trying to use url or header getters on them will throw. this error is different from engine itself having an error which is reported using the typical c return.

Does swift support algebraic types? ideally this would be typed as ResolutionOutcome = Endpoint | Error. And getters would be specified only on corresponding classes meaning that using getter on the wrong type is impossible

Reply:
Swift does support that we can define certain methods only on certain types. But this should not be an issue because C throws an invalid argument error. Is it possible that C returns Success (ResolutionOutcome was as exepected) but the byte_cursor length is 0?

Reply:
We dont have to mimic c interface completely in swift if swift language provides better constructs for what we want. 
From conceptual perspective output of resolution is (Endpoint(url, headers, optional properties) OR Error(message)), i.e. its algebraic sum type that swift supports through a slightly weird enum syntax.

Reply:
With that said. The api is already used by swift, so there might not be much incentive to rewrite interface

Reply:
Thanks a lot for the help, I have updated the API and error handling as discussed.